File: 2024/july/javaScript/22.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function(names, heights) {
        let n = names.length;
        let ans = [];
        let id = Array.from({length : n} , (_ , i)=> i);
        id.sort((a , b)=> heights[b]-heights[a]);
        for (let i = 0; i < n; ++i) ans.push(names[id[i]]);
        return ans;

};

File: 2024/july/java/22.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        int n = names.length;
        var ans = new String[n];
        var id = new Integer[n];
        for (int i = 0; i < n; ++i) id[i] = i;

        Arrays.sort(id , new Comparator<Integer>(){
            @Override
            public int compare(Integer a , Integer b){
                return heights[b] - heights[a];
            }
        });
        for (int i = 0; i < n; ++i) ans[i] = names[id[i]];
        return ans;

    }
}
